src/unix/newlib/mod.rs

```diff
@@ -10,30 +10,58 @@ cfg_if! {
 }
 
 cfg_if! {
-    if #[cfg(any(target_os = "espidf"))] {
+    if #[cfg(target_os = "horizon")] {
+        pub type dev_t = u32;
+        pub type off_t = i64;
+    } else {
         pub type dev_t = ::c_short;
-        pub type ino_t = ::c_ushort;
         pub type off_t = ::c_long;
-    } else if #[cfg(any(target_os = "vita"))] {
-        pub type dev_t = ::c_short;
-        pub type ino_t = ::c_ushort;
-        pub type off_t = ::c_int;
-    } else {
-        pub type dev_t = u32;
+    }
+}
+
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
         pub type ino_t = u32;
-        pub type off_t = i64;
+    } else if #[cfg(target_arch = "sparc")] {
+        pub type ino_t = ::c_ulong;
+    } else {
+        pub type ino_t = ::c_ushort;
     }
 }
 
 pub type fsblkcnt_t = u64;
 pub type fsfilcnt_t = u32;
 pub type id_t = u32;
-pub type key_t = ::c_int;
+
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
+        pub type key_t = ::c_int;
+    } else {
+        pub type key_t = ::c_long;
+    }
+}
+
 pub type loff_t = ::c_longlong;
-pub type mode_t = ::c_uint;
-pub type nfds_t = u32;
+
+cfg_if! {
+    if #[cfg(target_os = "horizon")] {
+        pub type mode_t = ::c_uint;
+        pub type nfds_t = u32;
+    } else {
+        pub type mode_t = u32;
+        pub type nfds_t = ::c_uint;
+    }
+}
+
 pub type nlink_t = ::c_ushort;
-pub type pthread_t = ::c_ulong;
+
+cfg_if! {
```

Comment:
Ditto for `pthread_t`?

Reply:
https://sourceware.org/git/?p=newlib-cygwin.git;a=blob;f=newlib/libc/include/sys/_pthreadtypes.h;h=75e9e1cbf71070e6ffef02ced33d0cd94ace0c14;hb=9e09d6ed83cce4777a5950412647ccc603040409#l32

`__XMK__` seems to be not defined for rust's `cfg`